### ISSUE TITLE
Remove max height from editable-text in item pane

### ISSUE THREAD

**Issue by dstillman** [zotero/zotero#3646]
https://forums.zotero.org/discussion/comment/453833#Comment_453833

I know a lot of work went into getting this working right, but there shouldn't actually be a max height on anything in the item pane. We always showed the full title and expanded extract before, so it's a usability regression for no real reason, and the nested scrollbars look bad.

**Comment by dstillman** [comment]
Abstract is already a collapsible section, so that seems fine?

**Comment by dstillman** [comment]
I guess the argument is that there's value in seeing a few lines? I'm not sure if that's actually the case.

**Comment by dstillman** [comment]
What did we even show before? One truncated line?

**Comment by abaevbog** [comment]
Yes, and it would expand all the way when you click on it

**Comment by abaevbog** [comment]
In the screenshot from the forums, I totally see that there's no reason to have both collapsible section and the scrollbar at the same time. You would expect one or the other, right?

**Comment by dstillman** [comment]
Yeah, I'm just not sure that was providing much value beyond knowing that there was an abstract. I suppose we could try to do something custom in the section header if we thought it was useful, but that can happen separately.

**Comment by dstillman** [comment]
Oh, I was thinking the Abstract header already had that, but I see that it doesn't. That's easy to fix, though.

**Comment by dstillman** [comment]
(And yes, more thoughts for Title changes incoming…)

**Comment by abaevbog** [comment]
Something that I saw in a random excel-like app is a long text field shows a "preview" of a multiline field (like 4-5 lines or so), and if you want to see the whole thing, you would click a small icon at the bottom right of the textfield, which opens a large popup that displays all the text. Using this as an inspiration, what if we add a tiny "expand" icon to the bottom right of title, abstract, etc., and when you tap on it, it will remove the height limit? It would allow one to see the entire field if needed, and would not require collapsing anything, so the "preview" of the abstract doesn't actually go away.

**Comment by dstillman** [comment]
We already have a mechanism for expanding long editable things on click for annotations in the reader sidebar if we wanted it, but I really don't think this is a problem. Scrolling is OK! We can trivially add an indication to the Abstract header — even the same truncated first line — to show that there's an abstract, which brings us back to exactly what we had before, except much more discoverable. Title we can discuss separately, but along with moving it (more on that later), I'd like to move Locate so that we get the full width, which will hopefully get us not so far off from the height we were taking up at half pane (i.e., with the Title label) in a smaller font before. So I think we can just do this and deal with those things separately.

**Comment by abaevbog** [comment]
As a start, I added a PR where `editable-text` has no `max-height` unless `limited-height` (there might be a better label for it) attribute is added. There, all the fields (but the title, the only field that currently has it) expand all the way. Correct me if I missed anything

**Comment by abaevbog** [comment]
> Re attribute name, we could expose --min-visible-lines and --max-visible-lines as min-lines and max-lines attributes

Great idea! `max-height` logic is not used anywhere for now but I thought that we'd want to keep it around for the future use cases. `max-lines` and `min-lines` attributes is a nice way to expose this functionality
